### PR TITLE
TW-105: avoid blink in avtar and name

### PR DESCRIPTION
**Pull request by sherlockvn**
https://github.com/linagora/twake-on-matrix/assets/43041967/41e3ea4e-92b1-4dc8-9f3b-f618b8b44409

### CODE REVIEW

File: lib/pages/chat_list/chat_list.dart [hunk 1]
```diff
@@ -20,6 +20,8 @@ import 'package:matrix/matrix.dart';
 import 'package:receive_sharing_intent/receive_sharing_intent.dart';
 import 'package:uni_links/uni_links.dart';
 import 'package:vrouter/vrouter.dart';
+import 'package:async/async.dart';
+
```
Comment (imGok) [inline lib/pages/chat_list/chat_list.dart:25]: why 2 lines ?

**Review by imGok — APPROVED** [review]
LGTM

File: lib/pages/chat_list/chat_list.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/chat_list/chat_list.dart:25]: yes. no need but it looks good :D

File: lib/pages/chat_list/chat_list_body.dart [hunk 2]
```diff
@@ -80,7 +80,7 @@ class ChatListViewBody extends StatelessWidget {
                   const SizedBox(height: 16),
                   FutureBuilder<Profile?>(
                     // ignore: unnecessary_cast
-                    future: (client.fetchOwnProfile() as Future<Profile?>).onError((e, s) => null),
+                    future: controller.fetchOwnProfile(client: client),
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list_body.dart:83]: how about the error case?

File: lib/pages/chat_list/chat_list.dart [hunk 3]
```diff
@@ -71,6 +73,21 @@ class ChatListController extends State<ChatList>
 
   String? activeSpaceId;
 
+  late final profileMemoizers = <Client?, AsyncMemoizer<Profile>>{};
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:76]: how about to put it in the `matrix.dart`?

File: lib/pages/chat_list/chat_list.dart [hunk 4]
```diff
@@ -71,6 +73,21 @@ class ChatListController extends State<ChatList>
...
+
+  Future<Profile?> fetchOwnProfile({required Client client}) {
+    if (!profileMemoizers.containsKey(client)) {
+      profileMemoizers[client] = AsyncMemoizer();
+    }
+    return profileMemoizers[client]!.runOnce(() async {
+      return await client.fetchOwnProfile();
+    });
+  }
```
Comment (hoangdat) [inline lib/pages/chat_list/chat_list.dart:85]: IMO, before your change, `client.fetchOwnProfile()` only call one. Why we need `AsyncMemoizer`?

**Review by imGok — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**